### PR TITLE
Add workflow scripts and refactor InputGenerator

### PR DESCRIPTION
**Pull request by ghutchis**
Currently does not properly update the QtGui molecule for changes or use the undoMolecule. I'll add a comment.

### CODE REVIEW

**Comment by ghutchis** [comment]
The 2nd half of the project (i.e., refactor the quantum scripts to use the new base class) is hard for me to do, since Molequeue won't link on Mac 10.12.

The new widget provides the central "form" and the updated quantum code will need to handle the preview text, etc.

File: avogadro/qtplugins/workflows/scripts/test.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
+"""
```
Comment (ghutchis) [inline avogadro/qtplugins/workflows/scripts/test.py]: Obviously I need a real workflow-centered test.

File: avogadro/qtgui/interfacescript.cpp [hunk 2]
```diff
@@ -145,7 +161,74 @@ void InputGenerator::reset()
...
+    Io::FileFormatManager &formats = Io::FileFormatManager::instance();
+    QScopedPointer<Io::FileFormat> format(formats.newFormatFromFileExtension(
+                                            "cjson"));
+    // convert the "cjson" field to a string
+    QJsonObject cjsonObj = obj["cjson"].toObject();
+    QJsonDocument doc(cjsonObj);
+    QString strCJSON(doc.toJson(QJsonDocument::Compact));
+    if (!strCJSON.isEmpty()) {
+      result = format->readString(strCJSON.toStdString(), mol);
+      // TODO: need to indicate changes to the QtGui Molecule
```
Comment (ghutchis) [inline avogadro/qtgui/interfacescript.cpp]: mol is a Core::Molecule as a parameter, and I cannot seem to cast to a QtGui::Molecule here - I always get NULL.

Ideas or fixes are *very* welcome

**Comment by cryos** [comment]
> The 2nd half of the project (i.e., refactor the quantum scripts to use the new base class) is hard for me to do, since Molequeue won't link on Mac 10.12.

I updated my Mac to Sierra to try and fix that issue, but it builds/links fine here.

**Comment by cryos** [comment]
This fails to compile here in the molequeue directory:

/home/marcus/src/openchemistry/avogadrolibs/avogadro/molequeue/inputgeneratorwidget.h:22:28: fatal error: inputgenerator.h: No such file or directory
 #include "inputgenerator.h"

**Comment by cryos** [comment]
This builds when turning off MoleQueue, if you want to merge this earlier then I would suggest simply leaving the inputgenerator.* files in place and refactoring in a future branch. I was able to compile by turning off MoleQueue, but I don't see any entries to test the functionality. 

I turned on INSTALL_TEST_SCRIPTS, but then spotted that the CMake variable was incorrect. I just pushed a commit to your branch that fixes it, and the scale operation seems to work as expected. It looks really nice.

**Comment by ghutchis** [comment]
I was intending to re-factor the input generators in a separate push - not sure why the files were missing, but it was easy to revert.

**Pull request merged**